plugins/azure/applicationGateway/agSecurityLoggingEnabled.js

```diff
@@ -0,0 +1,70 @@
+    recommended_action: 'Modify Application Gateway and add diagnostic settings for Access and Firewall Logs.',
+    link: 'https://learn.microsoft.com/en-us/azure/application-gateway/application-gateway-diagnostics',
+    apis: ['applicationGateway:listAll', 'diagnosticSettings:listByApplicationGateways'],
+
+    run: function(cache, settings, callback) {
+        const results = [];
+        const source = {};
+        const locations = helpers.locations(settings.govcloud);
+        async.each(locations.applicationGateway, (location, rcb) => {
+            const applicationGateways = helpers.addSource(cache, source,
+                ['applicationGateway', 'listAll', location]);
+
+            if (!applicationGateways) return rcb();
+
+            if (applicationGateways.err || !applicationGateways.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query Application Gateway: ' + helpers.addError(applicationGateways), location);
+                return rcb();
+            }
+
+            if (!applicationGateways.data.length) {
+                helpers.addResult(results, 0, 'No existing Application Gateway found', location);
+                return rcb();
+            }
+
+            applicationGateways.data.forEach(function(appGateway) {
+                if (!appGateway.id) return;
+                const diagnosticSettings = helpers.addSource(cache, source,
+                    ['diagnosticSettings', 'listByApplicationGateways', location, appGateway.id]);
+
+                if (!diagnosticSettings || diagnosticSettings.err || !diagnosticSettings.data) {
+                    helpers.addResult(results, 3, 'Unable to query Application Gateway diagnostics settings: ' + helpers.addError(diagnosticSettings), location, appGateway.id);
+                } else {
+
+                    var logs = diagnosticSettings.data[0] && diagnosticSettings.data[0].logs ? diagnosticSettings.data[0].logs : [];
+                    const allLogsEnabled = logs.some(log => log.categoryGroup === 'allLogs' && log.enabled);
+                    if (allLogsEnabled) {
+                        helpers.addResult(results, 0, 'Application Gateway has security logging enabled', location, appGateway.id);
+                    } else {
+                        const requiredLogs = ['ApplicationGatewayAccessLog', 'ApplicationGatewayFirewallLog'];
+                        const missingLogs = requiredLogs.filter(requiredCategory => 
+                            !logs.find(log => (log.category === requiredCategory && log.enabled))
+                        );
+                        
+                        if (missingLogs.length) {
```

Comment:
```suggestion
                        if (!allLogsEnabled && missingLogs.length) {
```
why don't we just add the check for allLogsEnabled here instead of adding a separate if?